### ISSUE TITLE
Dot Product should be normalized.

### ISSUE THREAD

**Issue by aneesh-joshi** [faneshion/MatchZoo#166]
https://github.com/faneshion/MatchZoo/blob/e8209d39933a737db693d926df20c498ce30334b/matchzoo/models/drmm_tks.py#L80

Since we are training with `1` and `0` as labels, the Dot product should have `normalize=True` 

What do you think @faneshion @bwanglzu @yangliuy

**Comment by bwanglzu** [comment]
It's an un-published paper..have to ask @faneshion for more information.

**Comment by faneshion** [comment]
It all depends on the loss function. If you take the pair hinge loss `max(0, theta - s0 + s1)`, the output can be a score. By the way, for some loss, such as cross-entropy, the output of the model should be a probability.

**Comment by aneesh-joshi** [comment]
@faneshion 
Ok
That makes sense.

On a side note:
I was building my own implementation of drmm_tks using MatchZoo like code.
Oddly enough, Mean Squared Error is providing better ndcg and map scores on wikiqa as compared to rank hinge loss.

Would you have any idea as to why this is happening?
